### PR TITLE
Add metadata versions v9 - v14

### PR DESCRIPTION
**Pull request by justkawal**
## Description
* Adds metadata version schema.
* added linter rule to allow cases insensitive names to pass through CI.
* Added OldTypes class to hold metadata-versions.

### CODE REVIEW

File: packages/substrate_metadata/lib/old/definitions/metadata/base.dart [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+/// Referenced from: https://github.com/subsquid/squid/blob/%40subsquid/substrate-metadata_v1.1.0/substrate-metadata/src/old/definitions/metadata/base.ts
+
+part of metadata;
+
+const Si1Variant = <String, String>{
+  'name': 'Text',
+  'fields': 'Vec<Si1Field>',
+  'index': 'u8',
+  'docs': 'Vec<Text>'
+};
```
Comment (Lohann) [inline packages/substrate_metadata/lib/old/definitions/metadata/base.dart:12]:
```dart
// ignore: non_constant_identifier_names
const Si1Variant = <String, String>{
  'name': 'Text',
  'fields': 'Vec<Si1Field>',
  'index': 'u8',
  'docs': 'Vec<Text>'
};
```

File: packages/substrate_metadata/analysis_options.yaml [hunk 2]
```diff
@@ -13,6 +13,10 @@
 
 include: package:lints/recommended.yaml
 
+linter:
+  rules:
+    constant_identifier_names: false
```
[the remark this answers is not in the document]
Comment (justkawal) [reply packages/substrate_metadata/analysis_options.yaml:18]: Alright, I'll change it right-away.

File: packages/substrate_metadata/lib/old/definitions/metadata/base.dart (hunk 1 above)
Comment (justkawal) [reply packages/substrate_metadata/lib/old/definitions/metadata/base.dart:12]: Alright, I'll change it right-away.

File: packages/substrate_metadata/lib/old/definitions/metadata/base.dart [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+const ScaleInfoTypes = <String, dynamic>{
```
[the remark this answers is not in the document]
Comment (justkawal) [reply packages/substrate_metadata/lib/old/definitions/metadata/base.dart:14]: Alright, I'll change it right-away.

**Review by Lohann — APPROVED** [review]

**Review by leonardocustodio — APPROVED** [review]

**Review by leonardocustodio — DISMISSED** [review]
Do we need this `OldTypes` maybe we can think a better name?

File: packages/substrate_metadata/lib/old/types.dart [hunk 4]
```diff
@@ -0,0 +1,55 @@
+class OldTypes {
```
Comment (pastre) [inline packages/substrate_metadata/lib/old/types.dart:1]: @leonardocustodio  just brainstorming names, not all of them make sense
```suggestion
class LegacyTypes {
```
```suggestion
class UnusedTypes {
```

```suggestion
class UnsupportedTypes {
```

```suggestion
class ThrowbackTypes {
```

Comment (justkawal) [reply packages/substrate_metadata/lib/old/types.dart:1]: @pastre `LegacyTypes` sounds good to me.

File: packages/substrate_metadata/lib/old/legacy_types_model.dart [hunk 5]
```diff
@@ -0,0 +1,55 @@
+class LegacyTypes {
```
Comment (pastre) [inline packages/substrate_metadata/lib/old/legacy_types_model.dart:1]: Shouldn't we add tests for these classes?

Comment (pastre) [reply packages/substrate_metadata/lib/old/legacy_types_model.dart:1]: Also, please prefer [composition over inheritance](https://en.wikipedia.org/wiki/Composition_over_inheritance). I think a [decorator ](https://refactoring.guru/design-patterns/decorator) is a much scalable solution to this problem

Comment (justkawal) [reply packages/substrate_metadata/lib/old/legacy_types_model.dart:1]: > Shouldn't we add tests for these classes?

It's just a model class, there is no need to test a static sitting model class.

Please check big repositories like: **parity_scale_codec** and then refer whether we need to write tests for a model class or not !!

Comment (justkawal) [reply packages/substrate_metadata/lib/old/legacy_types_model.dart:1]: As I've had a discussion with @Lohann earlier, that we should focus on testing the Logics more.

Comment (justkawal) [reply packages/substrate_metadata/lib/old/legacy_types_model.dart:1]: @pastre Please try to understand we're not seeking for `100%` code coverage for now. We gotta make it simple, strong, fast and quick before the deadline.

File: packages/substrate_metadata/lib/old/legacy_types_model.dart [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+  LegacyTypesWithSpecVersionRange(
+      {required this.minmax,
+      super.types,
+      super.typesAlias,
+      super.signedExtensions});
+
+  static LegacyTypesWithSpecVersionRange fromMap(Map<String, dynamic> map) {
+    return LegacyTypesWithSpecVersionRange(
+        minmax: (map['minmax'] as List).cast<int?>(),
```
Comment (pastre) [inline packages/substrate_metadata/lib/old/legacy_types_model.dart:28]: This is logic and needs to be tested. If this is supposed to only be only a data model, please consider creating a separate class to map your data types

**Review by pastre — DISMISSED** [review]

**Comment by pastre** [comment]
I'm happy with an issue for those comments :)

File: packages/substrate_metadata/lib/old/legacy_types_model.dart [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    var obj = LegacyTypesBundle(
+        types: map['types'],
+        typesAlias: map['typesAlias'],
+        signedExtensions: map['signedExtensions']);
+
+    if (map['versions'] != null) {
+      obj.versions = (map['versions'] as List)
+          .map((value) => LegacyTypesWithSpecVersionRange.fromMap(value))
+          .toList();
+    }
```
[the remark this answers is not in the document]
Comment (justkawal) [reply packages/substrate_metadata/lib/old/legacy_types_model.dart:51]:
```
TODO: Close issue: #110 to resolve this.
```

File: packages/substrate_metadata/lib/old/legacy_types_model.dart (hunk 6 above)
Comment (justkawal) [reply packages/substrate_metadata/lib/old/legacy_types_model.dart:28]: TODO: Close issue: #110 to resolve this.

**Review by gabrielokura — APPROVED** [review]

**Pull request merged**